File: run_migration.mjs
#!/usr/bin/env node
/**
 * Script untuk menjalankan SQL migration ke Supabase
 * Usage: node run_migration.mjs
 */

import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

// Load environment variables from .env
function loadEnv() {
  const envPath = path.join(__dirname, '.env');
  const envContent = fs.readFileSync(envPath, 'utf-8');
  const env = {};
  
  envContent.split('\n').forEach(line => {
    line = line.trim();
    if (line && !line.startsWith('#')) {
      const [key, ...rest] = line.split('=');
      const value = rest.join('=').replace(/^"|"$/g, '');
      env[key] = value;
    }
  });
  
  return env;
}

// Execute SQL via Supabase REST API
async function executeSql(url, apiKey, sql) {
  const statements = sql
    .split(';')
    .map(s => s.trim())
    .filter(s => s.length > 0);

  console.log(`\n📊 Total ${statements.length} SQL statements akan dijalankan\n`);
  console.log('⚠️  REST API tidak mendukung DDL statements langsung.');
  console.log('📋 Silakan copy-paste SQL berikut ke Supabase SQL Editor:\n');
  console.log('='.repeat(80));
  console.log(sql);
  console.log('='.repeat(80));
  console.log('\n📍 Langkah:');
  console.log('1. Buka https://app.supabase.com');
  console.log('2. Pilih project rapor-cepat');
  console.log('3. Buka SQL Editor');
  console.log('4. Buat query baru dan paste SQL di atas');
  console.log('5. Klik tombol RUN');
}

async function main() {
  try {
    const env = loadEnv();
    
    const supabaseUrl = env.VITE_SUPABASE_URL;
    const apiKey = env.VITE_SUPABASE_PUBLISHABLE_KEY;
    
    if (!supabaseUrl || !apiKey) {
      console.error('❌ Error: Environment variables tidak ditemukan di .env');
      process.exit(1);
    }
    
    console.log('🔗 Supabase Configuration:');
    console.log(`   URL: ${supabaseUrl}`);
    console.log(`   Key: ${apiKey.substring(0, 20)}...`);
    
    const sqlFile = path.join(__dirname, 'supabase/migrations/20251203084341_add_ekstrakurikuler_and_prestasi.sql');
    
    if (!fs.existsSync(sqlFile)) {
      console.error(`❌ Error: File ${sqlFile} tidak ditemukan`);
      process.exit(1);
    }
    
    const sql = fs.readFileSync(sqlFile, 'utf-8');
    console.log(`\n📝 SQL file ditemukan (${sql.length} bytes)\n`);
    
    await executeSql(supabaseUrl, apiKey, sql);
    
  } catch (error) {
    console.error('❌ Error:', error.message);
    process.exit(1);
  }
}

main();

File: run_migration.py
#!/usr/bin/env python3
"""
Script untuk menjalankan migration SQL ke Supabase
"""
import os
import sys
import requests
import json

# Baca environment variables dari .env
def load_env():
    env_vars = {}
    if os.path.exists('.env'):
        with open('.env', 'r') as f:
            for line in f:
                line = line.strip()
                if line and not line.startswith('#'):
                    key, value = line.split('=', 1)
                    env_vars[key] = value.strip('"')
    return env_vars

# Baca SQL file
def read_sql_file(filename):
    with open(filename, 'r') as f:
        return f.read()

# Jalankan SQL
def execute_sql(supabase_url, api_key, sql):
    """Jalankan SQL menggunakan Supabase REST API"""
    headers = {
        'Authorization': f'Bearer {api_key}',
        'Content-Type': 'application/json',
        'Prefer': 'return=representation'
    }
    
    # Split SQL into individual statements
    statements = [s.strip() for s in sql.split(';') if s.strip()]
    
    print(f"Total {len(statements)} SQL statements akan dijalankan")
    
    for i, statement in enumerate(statements, 1):
        print(f"\n[{i}/{len(statements)}] Menjalankan statement...")
        print(f"Query: {statement[:80]}...")
        
        try:
            # Gunakan endpoint rpc untuk menjalankan custom SQL
            response = requests.post(
                f'{supabase_url}/rest/v1/rpc/execute_sql',
                headers=headers,
                json={'query': statement}
            )
            
            if response.status_code in [200, 201]:
                print(f"✓ Berhasil")
            else:
                print(f"✗ Error: {response.status_code}")
                print(response.text)
                
        except Exception as e:
            print(f"✗ Exception: {str(e)}")

def main():
    env = load_env()
    
    supabase_url = env.get('VITE_SUPABASE_URL')
    api_key = env.get('VITE_SUPABASE_PUBLISHABLE_KEY')
    
    if not supabase_url or not api_key:
        print("❌ Error: VITE_SUPABASE_URL atau VITE_SUPABASE_PUBLISHABLE_KEY tidak ditemukan di .env")
        sys.exit(1)
    
    print(f"🔗 Supabase URL: {supabase_url}")
    print(f"🔑 API Key: {api_key[:20]}...")
    
    sql_file = 'supabase/migrations/20251203084341_add_ekstrakurikuler_and_prestasi.sql'
    
    if not os.path.exists(sql_file):
        print(f"❌ Error: File {sql_file} tidak ditemukan")
        sys.exit(1)
    
    sql = read_sql_file(sql_file)
    print(f"\n📝 SQL file ditemukan ({len(sql)} bytes)")
    
    execute_sql(supabase_url, api_key, sql)
    
    print("\n✅ Migration selesai!")

if __name__ == '__main__':
    main()
